src/main/java/car/domain/RacingCars.java

```diff
@@ -9,16 +10,17 @@
 
     private final Random random = new Random();
 
-    private List<RacingCar> racingCars = new ArrayList<>();
+    private final List<RacingCar> racingCars = new ArrayList<>();
 
     public List<RacingCar> getRacingCars() {
         return racingCars;
     }
 
-    public void addCars(int carCount) {
-        for (int i = 0; i < carCount; ++i) {
-            racingCars.add(new RacingCar());
-        }
+    public void addCars(String[] carNameList) {
+        Arrays.stream(carNameList)
+                .forEach( it ->
+                        racingCars.add(new RacingCar(it))
+                );
```

Comment:
```suggestion
    public void addCars(String[] carNameList) {
        for(String carName: carNameList) {
            racingCars.add(new RacingCar(carName));
        }
```

굳이 기존의 for 문을, forEach 로 변환하신 이유가 있으시나요?
제가 함수형 프로그래밍에 관해서 다른 분들께도 강조하는 내용이기도 하지만,
함수형 프로그래밍의 핵심은 함수형 인터페이스의 사용이 아니라, "불변" 객체를 갱신하는 것이 중요합니다.
(로버트 C 마틴의 클린 아키텍처를 읽어보시기 바랍니다.)

만약 정 함수형을 사용하고 싶다면, carNameList 를 배열이 아닌 List 자료형으로 통일해주시고,
아래와 같은 map 을 통해 racingCars 를 할당해주시면 좋을 것 같습니다.
racingCars 의 final 을 유지하기 위해, 아래의 로직은 생성자에게 들어가야만 하겠군요.
(생성자에 로직을 넣지 말라는 포비님 강의에 위배되기는 하네요.)

```
        racingCars = carNameList.stream()
                .map(RacingCar::new)
                .collect(Collectors.toList());
```

본인이 판단해서 더 좋은 것을 사용하면 좋을 것 같습니다.
지금의 일반 배열을 굳이 stream 으로 변환하는 것은 성능적인 면에서든, 가독성의 면에서든 별로 좋을게 없어보입니다.